src/Datadog.Trace/Agent/Transports/HttpStreamResponse.cs

```diff
@@ -0,0 +1,37 @@
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Datadog.Trace.Agent.Transports
+{
+    internal class HttpStreamResponse : IApiResponse
+    {
+        public HttpStreamResponse(int statusCode, long contentLength, Encoding encoding, Stream responseStream)
+        {
+            StatusCode = statusCode;
+            ContentLength = contentLength;
+            Encoding = encoding;
+            ResponseStream = responseStream;
+        }
+
+        public int StatusCode { get; }
+
+        public long ContentLength { get; }
+
+        public Encoding Encoding { get; }
+
+        public Stream ResponseStream { get; }
+
+        public void Dispose()
+        {
+        }
+
+        public async Task<string> ReadAsStringAsync()
+        {
+            using (var reader = new StreamReader(ResponseStream, Encoding, detectEncodingFromByteOrderMarks: false, 10240, leaveOpen: true))
```

Comment:
I wonder how do you pick that buffer size magic number, I guess it' the response length of 90% of cases or something like that.

Reply:
It's just an arbitrary "big enough" number.
But I checked the size of the response from the agent and it's only 148 bytes, so something much smaller may be ideal.
Also, memory stream does have a Length property, but it is a `long`.
Do you think this is worth checking?

Reply:
I did reduce the size for now, I think this is worth testing in smaller changesets.

Reply:
I consolidated the buffer size into `DatadogHttpClient`, but I'm happy with it living anywhere else.